IRM5

thbanasi opened the pull request:
Fixes #

## Proposed Changes

  -
  -
  -

shainw commented on Solutions/InsiderRiskManagement/Analytic Rules/InsiderM365IRMAlertObserved.yaml:
```diff
@@ -0,0 +1,38 @@
+id: 69660e65-0e5c-4700-8b99-5caf59786606
+name: (Private Preview) Insider Risk - Microsoft 365 Insider Risk Management Alert Observed
```
Do we want to put Private Preview here or just Preview?  We have traditionally used Preview for things like TI. :)

shainw commented on Solutions/InsiderRiskManagement/Analytic Rules/InsiderM365IRMAlertObserved.yaml:
```diff
@@ -0,0 +1,38 @@
+  | extend UserPrincipalName = UPN
+  | distinct AlertName, ProductName, Status, AlertLink, UserPrincipalName, Tactics, TimeGenerated
+  | sort by TimeGenerated desc
+  | limit 250
```
Since this is a "Detection" meaning it is an Analytics and not a Hunting query, it would make more sense to make sure a customer cannot get 250 results from this query as that would result in 250 alerts.  Can you look to tighten up the query or do aggregation that would help prevent this?

thbanasi replied:
Agreed, reduced

shainw commented on Solutions/InsiderRiskManagement/Analytic Rules/InsiderRiskyAccessByApplication.yaml:
```diff
@@ -0,0 +1,35 @@
+  | where AppDisplayName <> ""
+  | project Percent, UserPrincipalName, Location, AppDisplayName, RiskState
+  | sort by Percent desc
+  | limit 250
```
With evaluate, I doubt you would get 250, but same question as the previous query.  Can we mostly prevent high number of results?

shainw commented on Solutions/InsiderRiskManagement/Analytic Rules/InsiderRiskyAccessByApplication.yaml:
```diff
@@ -0,0 +1,35 @@
+  // | where Percent > 50 // Adjust & Tune Thresholds within Organzational Requirements
+  | where UserPrincipalName <> ""
+  | where AppDisplayName <> ""
+  | project Percent, UserPrincipalName, Location, AppDisplayName, RiskState
```
There is not time value passed thru, this is highly recommended.

thbanasi replied:
Added, mainly with arg_max, thanks

shainw commented on Solutions/InsiderRiskManagement/Hunting Queries/InsiderAnomalousAccess.yaml:
```diff
@@ -0,0 +1,58 @@
+      | summarize count() by UserPrincipalName
+      | project-rename UncommonAction = count_;
+  let Uncommon = UncommonActionVolume
+      | join(UncommonAction) on UserPrincipalName;
```
Please include join kinds with the query, LA will notify you of the recommendation and it is highly recommended.  Example - | join kind = inner

shainw commented on Solutions/InsiderRiskManagement/Hunting Queries/InsiderAnomalousAccess.yaml:
```diff
@@ -0,0 +1,58 @@
+  | join kind=fullouter(FirstTime) on UserPrincipalName
+  | where UserPrincipalName <> ""
+  | where FirstTimeDeviceLogon > 3 // Adjust & Tune Thresholds within Organizational Requirements
+  | project
```
No time value projected, highly recommended that this is included.

shainw reviewed (changes requested):
I commented on the first couple, the same comments apply througout the other queries.  Also, because some of these are effectively the same query, but with a slight modification in what is being counted and the threshold, we should look at using custom alert descriptions and names to build 1 query that supports the 5 types of items you want to alert on based on different thresholds.  Have a look at the [customdetails](https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#customdetails) and [alertdetailsoverride](https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#alertdetailsoverride) features.  Ashwin can also help on this if you have questions.

aprakash13 commented:
This is waiting on PR#3085. Once it is approved, please make the suggested changes above and we can go ahead and merge these. Thanks TJ.

aprakash13 reviewed (changes requested):
A few changes that I guess I had missed posting

thbanasi commented on Solutions/InsiderRiskManagement/Analytic Rules/InsiderHighUserAlertsCorrelation.yaml:
```diff
@@ -0,0 +1,85 @@
+  - entityType: Account
+    fieldMappings:
+      - identifier: AadUserId
+        columnName: UPN
```
Updated, thanks

thbanasi commented on Solutions/InsiderRiskManagement/Analytic Rules/InsiderRiskSensitiveDataAccessOutsideOrgGeo.yaml:
```diff
@@ -0,0 +1,41 @@
+  - entityType: Account
+    fieldMappings:
+      - identifier: AadUserId
+        columnName: UserPrincipalName
```
Updated, thanks

thbanasi commented on Solutions/InsiderRiskManagement/Analytic Rules/InsiderHighUserIncidentsCorrelation.yaml:
```diff
@@ -0,0 +1,85 @@
+  - entityType: Account
+    fieldMappings:
+      - identifier: AadUserId
+        columnName: UPN
```
Updated, thanks

v-rucdu commented:
@aprakash13 : Could you please take a look on the fixes that thbanasi  have made... Thanks!!

thbanasi commented:
Ajeet,
Thanks for the update and for your support. I'll be updating these rules and resubmitting by COB tomorrow.

Very Respectfully,

TJ Banasik
CISSP-ISSEP,ISSAP,ISSMP
Senior Program Manager
C&AI, Sovereign Cloud Security Group
***@***.******@***.***>
(202) 658-8148



***@***.***

LinkedIn: In/tjbanasik<http://www.linkedin.com/in/tjbanasik>

From: v-rucdu ***@***.***>
Sent: Thursday, September 30, 2021 3:28 AM
To: Azure/Azure-Sentinel ***@***.***>
Cc: TJ Banasik ***@***.***>; Author ***@***.***>
Subject: Re: [Azure/Azure-Sentinel] IRM5 (#3045)


@aprakash13<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Faprakash13&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C4cdee8fc21a040781e5008d983e3e3e6%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637685837076572796%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=kiyBiAeqWR2poZv5WXBCmwfOsNYWi0qyw4jGFehfcUY%3D&reserved=0> : Could you please take a look on the fixes that thbanasi have made... Thanks!!

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2FAzure-Sentinel%2Fpull%2F3045%23issuecomment-930924279&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C4cdee8fc21a040781e5008d983e3e3e6%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637685837076582789%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=%2FPjzI5SEHdrQTGR6Hf398SBwzjAJZMuN7tL6xMU0fwA%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAM6PREWIQ5Q4ZTSENXBQAOTUEQGRTANCNFSM5EDPXIZQ&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C4cdee8fc21a040781e5008d983e3e3e6%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637685837076592783%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=5ytLP4ekYEY70m%2FvZsyJNSq%2F1XWYCmBTGJYBr%2FFzMSA%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C4cdee8fc21a040781e5008d983e3e3e6%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637685837076592783%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=XjshT850SLWlnRaaW76rc%2FQsZ0OheNkEoieEbpUKVZs%3D&reserved=0> or Android<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C4cdee8fc21a040781e5008d983e3e3e6%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637685837076602780%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=iM5fzEh1oQGjG64xzoc%2FJynHbqS3yX3U0Sb1UsB6P1g%3D&reserved=0>.

thbanasi commented:
Ajeet,

I've Updated PR#3045 with the new IRM Rules, Hunting, & Workbook. Please let me know if anything further is needed for Approval, thank you.

Very Respectfully,

TJ Banasik
CISSP-ISSEP,ISSAP,ISSMP
Senior Program Manager
C&AI, Sovereign Cloud Security Group
***@***.******@***.***>
(202) 658-8148



***@***.***

LinkedIn: In/tjbanasik<http://www.linkedin.com/in/tjbanasik>

From: TJ Banasik
Sent: Thursday, September 30, 2021 6:47 AM
To: Azure/Azure-Sentinel ***@***.***>; Azure/Azure-Sentinel ***@***.***>
Cc: Author ***@***.***>
Subject: RE: [Azure/Azure-Sentinel] IRM5 (#3045)

Ajeet,
Thanks for the update and for your support. I'll be updating these rules and resubmitting by COB tomorrow.

Very Respectfully,

TJ Banasik
CISSP-ISSEP,ISSAP,ISSMP
Senior Program Manager
C&AI, Sovereign Cloud Security Group
***@***.******@***.***>
(202) 658-8148



***@***.***

LinkedIn: In/tjbanasik<http://www.linkedin.com/in/tjbanasik>

From: v-rucdu ***@***.******@***.***>>
Sent: Thursday, September 30, 2021 3:28 AM
To: Azure/Azure-Sentinel ***@***.******@***.***>>
Cc: TJ Banasik ***@***.******@***.***>>; Author ***@***.******@***.***>>
Subject: Re: [Azure/Azure-Sentinel] IRM5 (#3045)


@aprakash13<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Faprakash13&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C4cdee8fc21a040781e5008d983e3e3e6%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637685837076572796%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=kiyBiAeqWR2poZv5WXBCmwfOsNYWi0qyw4jGFehfcUY%3D&reserved=0> : Could you please take a look on the fixes that thbanasi have made... Thanks!!

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2FAzure-Sentinel%2Fpull%2F3045%23issuecomment-930924279&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C4cdee8fc21a040781e5008d983e3e3e6%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637685837076582789%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=%2FPjzI5SEHdrQTGR6Hf398SBwzjAJZMuN7tL6xMU0fwA%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAM6PREWIQ5Q4ZTSENXBQAOTUEQGRTANCNFSM5EDPXIZQ&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C4cdee8fc21a040781e5008d983e3e3e6%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637685837076592783%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=5ytLP4ekYEY70m%2FvZsyJNSq%2F1XWYCmBTGJYBr%2FFzMSA%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C4cdee8fc21a040781e5008d983e3e3e6%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637685837076592783%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=XjshT850SLWlnRaaW76rc%2FQsZ0OheNkEoieEbpUKVZs%3D&reserved=0> or Android<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C4cdee8fc21a040781e5008d983e3e3e6%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637685837076602780%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=iM5fzEh1oQGjG64xzoc%2FJynHbqS3yX3U0Sb1UsB6P1g%3D&reserved=0>.

aprakash13 commented:
@thbanasi Please resolve the comments that Shain had suggested if you have already addressed those. Lets also chat about this offline.

thbanasi commented:
@Ajeet Prakash ***@***.***>,

I made the updates per Shain's thoughts and our sync including First Observed dates for everything except the anomalies rollup rules. Is something missing to approve the PR?

Very Respectfully,

TJ Banasik
CISSP-ISSEP,ISSAP,ISSMP
Senior Program Manager
C&AI, Sovereign Cloud Security Group
***@***.******@***.***>
(202) 658-8148



***@***.***

LinkedIn: In/tjbanasik<http://www.linkedin.com/in/tjbanasik>

From: aprakash13 ***@***.***>
Sent: Wednesday, October 6, 2021 4:29 AM
To: Azure/Azure-Sentinel ***@***.***>
Cc: TJ Banasik ***@***.***>; Mention ***@***.***>
Subject: Re: [Azure/Azure-Sentinel] IRM5 (#3045)


@thbanasi<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fthbanasi&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C580abf2852104d722bfe08d988a347c1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637691057159811398%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=r69J0o1NxyWYTL1MDZuERPWcwDFn5rTpa9E64TFHS4Q%3D&reserved=0> Please resolve the comments that Shain had suggested if you have already addressed those. Lets also chat about this offline.

-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2FAzure-Sentinel%2Fpull%2F3045%23issuecomment-935740006&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C580abf2852104d722bfe08d988a347c1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637691057159811398%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=CEPn7GJrrZMKWXcRv3I%2FmJfU8O1CVa88TwpnvmwD7d0%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAM6PREQ3CQRA6P2CQOTSJNTUFQCC5ANCNFSM5EDPXIZQ&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C580abf2852104d722bfe08d988a347c1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637691057159821392%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=6XQH26QLKmr5aTRiwpXZGpd%2BqxlYatpNLtmcNh3omsw%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C580abf2852104d722bfe08d988a347c1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637691057159821392%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=I7uRBZ4kBF%2BUAbPzNCLthgITaReODPBZWOmxGYbD3mU%3D&reserved=0> or Android<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7CThomas.Banasik%40microsoft.com%7C580abf2852104d722bfe08d988a347c1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637691057159831386%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Tytr0po0IebUb0FZk7xXzgcvoPB51YhHbXZEdDKcJ00%3D&reserved=0>.

thbanasi commented:
Team, I've rewritten all (10) Insider Risk Management Analytics Rules and Hunting Queries. I've incorporated all the changes recommended by Shain, Ashwin, and Ajeet (MSTIC) to include specifying join types, adding detailed descriptions, including time thresholds, including make_set for context, referencing previous alert/incident links, etc. I've shifted focus to include options to correlate Azure Sentinel watchlist against all rules, it's commented as optional as this is a key requirement for insider threat teams. There is also an improved correlation to M365 IRM alerts to demonstrate the Azure Sentinel <> IRM better together story. Lastly, the Hunting Queries were to repetitive. I've adjusted the first per Ajeet's recommended updates. I deleted and rewrote the remaining 4 hunting queries for insider scenarios highlighted in IRM's insider scenario documentation (exfiltration, sabotage, elevating access, anomalous access etc). The PR now passes all validation checks, would it be possible to approve so we can include in the Solution Preview.

thbanasi commented:
@v-jayakal , would it be possible to provide an update on what's needed to approve this PR? There is an ongoing Preview and we need to get the new Workbooks and Rules to customers.
